src/rcms/fm/app/level1/HCALlevelTwoEventHandler.java

```diff
@@ -452,135 +457,109 @@ public void configureAction(Object obj) throws UserActionException {
-          functionManager.getHCALparameterSet().put(new FunctionManagerParameter<StringT>("CONFIGURED_WITH_RUN_KEY",new StringT(RunKey)));
-        }
-        else {
-          String warnMessage = "[HCAL LVL2 " + functionManager.FMname + "] Did not receive a run key.\nThis is probably OK for normal HCAL LVL2 operations ...";
-          logger.warn(warnMessage);
-        }
+        try{
+          // Determine the run type from the configure command
+          CheckAndSetParameter(       parameterSet, "HCAL_RUN_TYPE" );
+          CheckAndSetTargetParameter( parameterSet, "HCAL_RUN_TYPE" ,"CONFIGURED_WITH_RUN_KEY",true);
 
-        // get the tpg key from the configure command
-        if (parameterSet.get("TPG_KEY") != null) {
-          TpgKey = ((StringT)parameterSet.get("TPG_KEY").getValue()).getString();
-          functionManager.getHCALparameterSet().put(new FunctionManagerParameter<StringT>("CONFIGURED_WITH_TPG_KEY",new StringT(TpgKey)));
-          String warnMessage = "[HCAL LVL2 " + functionManager.FMname + "] Received a L1 TPG key: " + TpgKey;
-          logger.warn(warnMessage);
-        }
-        else {
-          String warnMessage = "[HCAL LVL2 " + functionManager.FMname + "] Did not receive a L1 TPG key.\nThis is only OK for HCAL local run operations ...";
-          logger.warn(warnMessage);
-        }
+          // Check and receive TPG key
+          CheckAndSetTargetParameter( parameterSet, "TPG_KEY" ,"CONFIGURED_WITH_TPG_KEY",true);
 
-        // get the info from the LVL1 if special actions due to a central CMS clock source change are indicated
-        ClockChanged = false;
-        if (parameterSet.get("CLOCK_CHANGED") != null) {
-          ClockChanged = ((BooleanT)parameterSet.get("CLOCK_CHANGED").getValue()).getBoolean();
-          functionManager.getHCALparameterSet().put(new FunctionManagerParameter<BooleanT>("CLOCK_CHANGED",new BooleanT(ClockChanged)));
+          // get the info from the LVL1 if special actions due to a central CMS clock source change are indicated
+          ClockChanged = false;
+          CheckAndSetParameter(       parameterSet, "CLOCK_CHANGED" );
+          ClockChanged = ((BooleanT)functionManager.getHCALparameterSet().get("CLOCK_CHANGED").getValue()).getBoolean();
           if (ClockChanged) {
-            logger.warn("[HCAL LVL2 " + functionManager.FMname + "] Did receive a request to perform special actions due to central CMS clock source change during the configureAction().\nThe ClockChange is: " + ClockChanged);
-          }
-          else {
-            logger.debug("[HCAL LVL2 " + functionManager.FMname + "] Did not receive a request to perform special actions due to central CMS clock source change during the configureAction().\nThe ClockChange is: " + ClockChanged);
+            logger.warn("[HCAL LVL2 " + functionManager.FMname + "] Did receive a request to perform special actions due to central CMS clock source change during the configureAction().");
```

Comment:
I don't see `ClockChanged` being used anywhere in the code. Keeping this warning message only in case we get a `CLOCK_CHANGED` from LV0